Content.Server/Objectives/Components/StealConditionComponent.cs

```diff
@@ -1,28 +1,64 @@
 using Content.Server.Objectives.Systems;
+using Content.Shared.Objectives;
 using Robust.Shared.Prototypes;
-using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
 
 namespace Content.Server.Objectives.Components;
 
 /// <summary>
-/// Requires that you steal a certain item.
+/// Requires that you steal a certain item (or several)
 /// </summary>
 [RegisterComponent, Access(typeof(StealConditionSystem))]
 public sealed partial class StealConditionComponent : Component
 {
     /// <summary>
-    /// The id of the item to steal.
+    /// A group of items to be stolen
     /// </summary>
-    /// <remarks>
-    /// Works by prototype id not tags or anything so it has to be the exact item.
-    /// </remarks>
-    [DataField(required: true, customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>)), ViewVariables(VVAccess.ReadWrite)]
-    public string Prototype = string.Empty;
+    [DataField(required: true)]
+    public ProtoId<StealTargetGroupPrototype> StealGroup;
+
+    /// <summary>
+    /// When enabled, disables generation of this target if there is no entity on the map (disable for objects that can be created mid-round).
+    /// </summary>
+    [DataField]
+    public bool VerifyMapExistance = true;
+
+    /// <summary>
+    /// If the target may be alive but has died, it will not be counted
+    /// </summary>
+    [DataField]
+    public bool CheckAlive = false;
+    /// <summary>
+    /// The minimum number of items you need to steal to fulfill a objective
+    /// </summary>
+    [DataField]
+    public int MinCollectionSize = 1;
+
+    /// <summary>
+    /// The maximum number of items you need to steal to fulfill a objective
+    /// </summary>
+    [DataField]
+    public int MaxCollectionSize = 1;
+
+    /// <summary>
+    /// Target collection size after calculation
+    /// </summary>
+    [DataField]
+    public int CollectionSize;
 
     /// <summary>
     /// Help newer players by saying e.g. "steal the chief engineer's advanced magboots"
     /// instead of "steal advanced magboots. Should be a loc string.
     /// </summary>
-    [DataField("owner"), ViewVariables(VVAccess.ReadWrite)]
+    [DataField]
```

Comment:
How does it work? The datafield should be `ownerText` for YAML, but all the protos are still using `owner`

Reply:
oopsie